### PR TITLE
Add info for VDEV state and guid

### PR DESCRIPTION
**Pull request by jgrund**
To maintain parity with ZED info, add
VDEV state and guid fields.

This will tell us the current state of a VDEV

`ONLINE` etc...

Signed-off-by: Joe Grund <joe.grund@intel.com>

### CODE REVIEW

File: libzfs-sys/src/lib.rs [hunk 1]
```diff
@@ -54,15 +54,43 @@ pub fn zpool_config_hostname() -> String {
...
 }
 
+pub fn to_vdev_stat(mut xs: Vec<u64>) -> vdev_stat_t {
+    xs.shrink_to_fit();
+
+    unsafe { std::ptr::read(xs.as_ptr() as *const _) }
+}
 
+pub fn to_vdev_state(n: u32) -> Option<vdev_state_t> {
+    if n <= 7 {
```
Comment (johnsonw) [inline libzfs-sys/src/lib.rs]: What's the significance of the number 7?

File: libzfs-sys/src/lib.rs [hunk 2]
```diff
@@ -54,15 +54,43 @@ pub fn zpool_config_hostname() -> String {
...
+        Some(unsafe { std::mem::transmute(n) })
+    } else {
+        None
+    }
+}
+
+pub fn to_vdev_aux(n: u32) -> Option<vdev_aux_t> {
+    if n <= 18 {
```
Comment (johnsonw) [inline libzfs-sys/src/lib.rs]: What does 18 mean?

File: libzfs-sys/src/lib.rs (hunk 1 above)
Comment (jgrund) [reply libzfs-sys/src/lib.rs]: This function maps an `u32` to an int enum.

`7` is the upper bound, so if we have a number > 7 it won't fit in the enum.

File: libzfs-sys/src/lib.rs (hunk 2 above)
Comment (jgrund) [reply libzfs-sys/src/lib.rs]: This function maps an `u32` to an int enum.

`18` is the upper bound, so if we have a number > 18 it won't fit in the enum.

**Pull request merged**